airbyte-webapp/src/components/connection/CatalogTree/FieldRow.tsx

```diff
@@ -54,18 +54,32 @@ const FieldRowInner: React.FC<FieldRowProps> = ({
   const isCursor = equal(config?.cursorField, field.path);
   const isPrimaryKey = !!config?.primaryKey?.some((p) => equal(p, field.path));
   const isNestedField = SyncSchemaFieldObject.isNestedField(field);
+  const isDisabled = (isCursorEnabled && isCursor) || (isPrimaryKeyEnabled && isPrimaryKey) || isNestedField;
```

Comment:
Here's the culprit -- `isCursorEnabled` appears to actually be whether it's configurable, same with `isPrimaryKeyEnabled`.  Source-defined ones wouldn't be configurable, but are definitely required.

Reply:
Oh, good call! I did not have that down as a requirement, but of course it makes sense.

Reply:
Fixed this [here](https://github.com/airbytehq/airbyte/pull/20844/commits/b53c2a806cdb44f998600ca5d095cc3664702ca5). I updated the variable names to hopefully better express what they're used for. Thanks for catching that!